### PR TITLE
Add job name when results collate multiple jobs

### PR DESCRIPTION
**Pull request by fejta**
If `GcsPrefix: this-job,that-job` then ensure that the name config includes a `Job name` field (insert it to the front if not)

### CODE REVIEW

File: pkg/updater/read.go [hunk 1]
```diff
@@ -208,6 +209,50 @@ func readColumns(parent context.Context, client gcs.Downloader, group *configpb.
 	return cols[0:maxIdx], nil
 }
 
+type nameConfig struct {
```
Comment (chases2) [inline pkg/updater/read.go:212]: nameConfig seems like an important part of understanding this code, but I don't quite get what it is. It's a... "decompiled" representation of how test groups are named based on their config settings? Are the parts the properties to look for, or the properties themselves?

Some comments on this struct would help a lot.

File: util/gcs/read_test.go [hunk 2]
```diff
@@ -49,6 +49,42 @@ func link(path Path, name, other string) storage.ObjectAttrs {
...
 }
 
+func TestBuildJob(t *testing.T) {
+	cases := []struct {
+		path  string
+		build string
+		job   string
+	}{
+		{
+			path:  "gs://bucket/path/job/hello",
```
Comment (chases2) [inline util/gcs/read_test.go:59]: I'm starting to see a pattern here of files used by controller `foo` are generally put into a subfolder called `foo`. I like that organization.

**Review by chases2 — APPROVED** [review]
/hold
/lint

This does seem clearer, but I'm still double-guessing while reading this code. A few comments on unexported functions might help a lot.

File: pkg/updater/read.go (hunk 1 above)
Comment (fejta) [reply pkg/updater/read.go:212]: This struct is the same as before, now in a more appropriate location.

I generally think the best way to understand code -- especially package internal code -- is to read the code.

**Pull request merged**